### PR TITLE
Fix the issue of the addEventListener method being undefined during screen rotation on iOS browsers.

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #
https://github.com/cocos/cocos-engine/issues/17249

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: pal/screen-adapter/web/screen-adapter.ts [hunk 1]
```diff
@@ -474,8 +481,13 @@ class ScreenAdapter extends EventTarget {
...
                 notifyOrientationChange(tmpOrientation);
             };
-            mediaQueryPortrait.addEventListener('change', orientationChangeCallback);
-            mediaQueryLandscape.addEventListener('change', orientationChangeCallback);
+            if (mediaQueryPortrait.addEventListener) {
+                mediaQueryPortrait.addEventListener('change', orientationChangeCallback);
+                mediaQueryLandscape.addEventListener('change', orientationChangeCallback);
+            } else {
+                mediaQueryPortrait.addListener(orientationChangeCallback);
+                mediaQueryLandscape.addListener(orientationChangeCallback);
```
Comment (minggo) [inline pal/screen-adapter/web/screen-adapter.ts:489]: Should check addListener first?

File: pal/screen-adapter/web/screen-adapter.ts [hunk 2]
```diff
@@ -474,8 +481,13 @@ class ScreenAdapter extends EventTarget {
                 }
                 notifyOrientationChange(tmpOrientation);
             };
-            mediaQueryPortrait.addEventListener('change', orientationChangeCallback);
-            mediaQueryLandscape.addEventListener('change', orientationChangeCallback);
+            if (mediaQueryPortrait.addEventListener) {
+                mediaQueryPortrait.addEventListener('change', orientationChangeCallback);
+                mediaQueryLandscape.addEventListener('change', orientationChangeCallback);
```
Comment (minggo) [inline pal/screen-adapter/web/screen-adapter.ts:486]: Should also check it too?

Comment (bofeng-song) [reply pal/screen-adapter/web/screen-adapter.ts:486]: mediaQueryLandscape and mediaQueryPortrait are similar objects, so checking one of them is sufficien

**Review by minggo — APPROVED** [review]

**Comment by atrl** [comment]
这个还是会报错吧mediaQueryPortrait是 undefined 啊

**Comment by bofeng-song** [comment]
mediaQueryPortrait.addEventListener 已经加了判断，确认合并了这个pr 了？
哪个系统版本能复现，iphone 多少 @atrl

**Pull request merged**